zproject/urls.py

```diff
@@ -511,3 +512,14 @@
 # reverse url mapping points to i18n urls which causes the frontend
 # tests to fail
 urlpatterns = i18n_patterns(*i18n_urls) + urls + legacy_urls
+
+def handler400(request, exception):
+    # This behaves exactly like the default Django implementation in
+    # the case where you haven't made a template "400.html", which we
+    # haven't -- except that it doesn't call `@requires_csrf_token` to
+    # attempt to set a `csrf_token` variable that the template could
+    # use if there were a template.  Which is good because that
+    # codepath can raise an error on a bad request, which is exactly
+    # the case we're trying to handle when we get here.
+    return HttpResponseBadRequest(
+        '<h1>Bad Request (400)</h1>', content_type='text/html')
```

Comment:
Do we not need to do something to register this `handler400` method?

Reply:
Nope! Feels a little odd to me too, but documented here: https://docs.djangoproject.com/en/1.11/topics/http/views/#customizing-error-views
and indeed it seems to work, as tested by the test. That term "URLconf" in the docs isn't a link, but the term is defined here: https://docs.djangoproject.com/en/1.11/topics/http/urls/ and it means the module we call `zproject.urls`.

The no-registration model is at least consistent with how the rest of that module works, so it's the sensible place to put it if they're going to choose that style.